backend/src/queue/queue.service.ts

```diff
@@ -51,34 +51,23 @@ export class QueueService {
     ];
     const matchedUsers = await this.redisService.getAllKeys(namespaces);
 
-    if (this.isUserInQueue(matchedUsers, userId)) {
-      return null;
-    }
```

Comment:
Unfortunately we still need to perform this check using the user ID. There is a race condition that occurs when the new connection occurs and the other connection has not yet disconnected.

Reply:
Fixed